src/cyclonedx/agent/reportgenerator.php

```diff
@@ -0,0 +1,176 @@
+   *
+   * @param array $licenseData The license data.
+   * @return array The generated license.
+   */
+  public function createLicense(array $licenseData): array
+  {
+    return $this->generateLicense($licenseData);
+  }
+
+  /**
+   * Generates the report.
+   *
+   * @param array $bomData The BOM data.
+   * @return array The generated report.
+   */
+  public function generateReport($bomdata): array
+  {
+    $report = [
+      'bomFormat' => 'CycloneDX',
+      'schema' => 'http://cyclonedx.org/schema/bom-1.4.schema.json',
+      'specVersion' => '1.4',
+      'version' => 1.0,
+      'serialNumber' => 'urn:uuid:'. uuid_create(UUID_TYPE_TIME),
+      'metadata' => [
+        'timestamp' => date('c'),
+        'tools' => [
+          [
+            'vendor' => 'FOSSology',
+            'name' => 'FOSSology',
+            'version' => $bomdata['tool-version']
+          ]
+        ],
+        'component' => $bomdata['maincomponent']
+      ],
+      'components' => $bomdata['components']
+    ];
+
+    return $report;
+  }
+
+  /**
+   * Generates a component.
+   *
+   * @param array $componentData The component data.
+   * @return array The generated component.
+   */
+  private function generateComponent(array $componentData): array
+  {
+    $component = [
+      'type' => $componentData['type'],
+      'name' => $componentData['name']
+    ];
+
+    if (array_key_exists('mimeType', $componentData) && !empty($componentData['mimeType'])) {
+      $component['mime-type'] = $componentData['mimeType'];
+    }
+
+    if (array_key_exists('bomref', $componentData) && !empty($componentData['bomref'])) {
+      $component['bom-ref'] = $componentData['bomref'];
+    }
+
+    /**
+     * "Specifies the scope of the component. If scope is not specified,
+     *  'required' scope SHOULD be assumed by the consumer of the BOM."
+     */
+    if (array_key_exists('scope', $componentData) && !empty($componentData['scope'])) {
+      $component['scope'] = $componentData['scope'];
+    } else {
+      $component['scope'] = 'required';
+    }
+
+    if (array_key_exists('hashes', $componentData) && !empty($componentData['hashes'])) {
+      $component['hashes'] = $componentData['hashes'];
+    }
+
+    if (array_key_exists('licenses', $componentData) && !empty($componentData['licenses'])) {
+      $component['licenses'] = $componentData['licenses'];
+    }
+
+    if (array_key_exists('copyright', $componentData) && !empty($componentData['copyright'])) {
+      $component['copyright'] = $componentData['copyright'];
+    }
+
+    if (array_key_exists('description', $componentData) && !empty($componentData['description'])) {
+      $component['description'] = $componentData['description'];
+    }
+
+    return $component;
+  }
+
+  /**
+   * Generates a license.
+   *
+   * @param array $licenseData The license data.
+   * @return array The generated license.
+   */
+  private function generateLicense(array $licenseData): array
+  {
+    $license = [];
+
```

Comment:
CycloneDX schema does not support SPDX LicenseRef as valid license id. Proposing following changes to make LicenseRefs as a license expression.

Reply:
```diff
diff --git a/src/cyclonedx/agent/reportgenerator.php b/src/cyclonedx/agent/reportgenerator.php
index 90555bb4a..47d60049e 100644
--- a/src/cyclonedx/agent/reportgenerator.php
+++ b/src/cyclonedx/agent/reportgenerator.php
@@ -11,6 +11,8 @@
  */
 namespace Fossology\CycloneDX;
 
+use Fossology\Lib\Data\LicenseRef;
+
 class BomReportGenerator
 {
   /**
@@ -138,6 +140,13 @@ class BomReportGenerator
   {
     $license = [];
 
+    // Check license ID is a LicenseRef
+    if (array_key_exists('id', $licenseData) && !empty($licenseData['id']) &&
+      stripos($licenseData['id'], LicenseRef::SPDXREF_PREFIX) === 0) {
+      $license['expression'] = $licenseData['id'];
+      return $license;
+    }
+
     if (array_key_exists('id', $licenseData) && !empty($licenseData['id'])) {
       $license['license']['id'] = $licenseData['id'];
     } else if (array_key_exists('name', $licenseData) && !empty($licenseData['name'])) {
```

Reply:
Done